### DISCUSSION TITLE
Rustdesk macos application cannot be opened.

### DISCUSSION THREAD

**Discussion by cctyl** [rustdesk/rustdesk#5631] [General]
### Bug Description

I had previously installed the application, but after one launch, it couldn’t open. Attempted to delete it, but it indicated that it was still open and couldn’t be removed. So, I forcefully removed the application using the ‘rm -rf’ command. I then reinstalled version 1.2.2 of the application. However, I am still facing the same issue. I tried to start it using the ‘open’ command, but it still doesn’t work and provides the following error message:
![屏幕快照 2023-09-08 上午8 06 50](https://github.com/rustdesk/rustdesk/assets/34262992/fff82f7b-a543-4511-86d3-c4db0a560190)
![屏幕快照 2023-09-08 上午8 07 12](https://github.com/rustdesk/rustdesk/assets/34262992/130a5587-36b4-409c-8554-1511a3671ce3)

### How to Reproduce

just install it

### Expected Behavior

Under normal circumstances, it should start up and display the interface.

### Operating system(s) on local side and remote side

osx 10.14.6(It happened before it was started.) 

### RustDesk Version(s) on local side and remote side

1.2.2

### Screenshots

![屏幕快照 2023-09-08 上午8 06 50](https://github.com/rustdesk/rustdesk/assets/34262992/fff82f7b-a543-4511-86d3-c4db0a560190)
![屏幕快照 2023-09-08 上午8 07 12](https://github.com/rustdesk/rustdesk/assets/34262992/130a5587-36b4-409c-8554-1511a3671ce3)

**Comment by rustdesk** [comment]
run with sudo 
```
launchctl unload -w /Library/LaunchDaemons/com.carriez.RustDesk_service.plist
/bin/rm /Library/LaunchDaemons/com.carriez.RustDesk_service.plist
/bin/rm /Library/LaunchAgents/com.carriez.RustDesk_server.plist
```

then run

```
launchctl remove com.carriez.RustDesk_server
```

**Reply by cctyl** [reply]
Okay, after executing the above command, everything is running smoothly now. Thank you.

**Reply by Shekstvil** [reply]
What password is it wanting me to enter? When I paste these commands?

**Reply by rustdesk** [reply]
Your macos account password

**Reply by robertmylne** [reply]
Tried everything, this was the only thing that actually allowed me to open ruskdesk again.

**Reply by SkyExploreWasTaken** [reply]
Hey man ! I tried your solution, but the file doesn't exist at /Library/LaunchDaemons/, neither at /Library/LaunchAgents/, but RustDesk still can't open. Do you have an idea of what could be the problem ?

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/5825 https://github.com/rustdesk/rustdesk/issues/3369

**Comment by rustdesk** [comment]
Need refactor Mac --server/--service processes to fix this completely.

Why having this issue:
Currently `--server` does listen to this delegate event, but `--service` (run as launch daemon as system process) process does not. When both `--server`/`--service` process running, it is fine, `--server` (run as launch agent as user process)process will receive this delegate event and open RustDesk correctly, but if `--server` does not run correctly, `--service` runs, we will have this issue. So far no idea why `--server` fails.

https://github.com/rustdesk/rustdesk/blob/230eb7653211d5f6199c9dbe3f9896be2fe9597c/flutter/macos/Runner/AppDelegate.swift#L14

https://github.com/rustdesk/rustdesk/blob/230eb7653211d5f6199c9dbe3f9896be2fe9597c/src/platform/macos.rs#L621

Two possible solution.

1, add handling the delegate event in `--service`, but `--service` run as system, not sure if can open rustdesk as user
2, compile a seperate executable for `--service` since it is very simple, this way, `--service` will not capture the delegate.

**Reply by rustdesk** [reply]
Have a workaround of this, https://github.com/rustdesk/rustdesk/commit/81aec8402b053fcc0965412c2ae5ba09e34f03f4 & https://github.com/rustdesk/rustdesk/commit/85f387bfb9ec2a8067d3273829a21b1deb07efd4 & https://github.com/rustdesk/rustdesk/commit/5aec5b084ac36665dfac105348a9e5d56cced9e1
checking start time of --service/--server, will restart if --service starts earlier than --server (but still see `can not open` seldomely)

**Reply by rustdesk** [reply]
I have attempted to write a helper program for `--service`, 
https://github.com/rustdesk/rustdesk/commit/d407dfed0a18289a77acf1cc3211c32eeec08fde
However, failed to noterize it since it is not runtime hardened.
BTW, I am not sure if the helper will trap the delegate event since it locates in app, sharing the same bundle id. We may have to copy it out from the app, put in `/Library/PrivilegedHelperTools` like the others.

**Comment by SachaRoland** [comment]
Hey, 
I have downloaded Rustdesk 1.3.2 on a Mac Monterey but cannot open it or at least get the display started. 
I have tried the basics: restarting the computer, uninstalling-reinstalling, nothing change. 
I gave all permissions for the app to open but nothing happens when I click on it. 

Any tip please???
